core/src/engine/coro/pool.hpp

```diff
@@ -111,39 +126,74 @@ Pool<Task>::~Pool() = default;
 
 template <typename Task>
 typename Pool<Task>::CoroutinePtr Pool<Task>::GetCoroutine() {
-  struct CoroutineMover {
-    std::optional<Coroutine>& result;
-
-    CoroutineMover& operator=(Coroutine&& coro) {
-      result.emplace(std::move(coro));
-      return *this;
-    }
-  };
-
   std::optional<Coroutine> coroutine;
-  CoroutineMover mover{coroutine};
 
   // First try to dequeue from 'working set': if we can get a coroutine
```

Comment:
It seems to me that he should stay here, since here they are looking at options for where to get the coroutine. But I added information about the local cache to the comment.

Reply:
On a second though, let's leave the old comment as it was before: it explains why we pool from `initial_coroutines_` after `used_coroutines_` and cache. The refactored comment tells what we do instead of why we do, let's just revert it